charts/opentelemetry-operator/README.md

```diff
@@ -0,0 +1,319 @@
+
+```console
+$ helm show values open-telemetry/opentelemetry-operator
+```
+
+## Install OpenTelemetry Collector
+
+_See [OpenTelemetry website](https://opentelemetry.io/docs/collector/) for more details about the Collector_
+
+Once the opentelemetry-operator deployment is ready, you can deploy OpenTelemetry Collector in our Kubernetes
+cluster.
+
+The Collector can be deployed as one of four modes: Deployment, DaemonSet, StatefulSet and Sidecar. The default
+mode is Deployment. We will introduce the benefits and use cases of each mode as well as giving an example for each.
+
+### Deployment Mode
+
+If you want to get more control of the OpenTelemetry collector and create a standalone application, Deployment would
+be your choice. With Deployment, you can relatively easily scale up the collector to monitor more targets, roll back
+to an early version if anything unexpected happens, pause the collector, etc. In general, you can manage your collector
+instance just as an application.
+
+The following example configuration deploys the Collector as Deployment resource. The receiver is jaeger-receiver and
+the exporter is logging exporter.
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: my-collector
+spec:
+  mode: deployment # This configuration is omittable.
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+    processors:
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+          processors: []
+          exporters: [logging]
+EOF
+```
+
+### DaemonSet Mode
+
+DaemonSet should satisfy your most basic needs and is the most common method to deploy OpenTelemetry collector.
+In this case, every Kubernetes node will have its own collector copy which would monitor the pods in it.
+
+The following example configuration deploys the Collector as DaemonSet resource. The receiver is jaeger-receiver and
+the exporter is logging exporter.
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: my-collector
+spec:
+  mode: daemonset
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+    processors:
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+          processors: []
+          exporters: [logging]
+EOF
+```
+
+### StatefulSet Mode
+If you want your collector to have stable persistent identities or storage, you should choose StatefulSet.
+Take Prometheus metrics for example, if you use above two approaches, the metrics collected from the receiver will
+be stored locally by default, which is ephemeral. However, StatefulSet allows you to configure a persistent storage
+interface called PersistentVolume. PersistentVolume will maintain the historical metrics data and even survive pods
```

Comment:
Persistent volumes can be used with deployments as well. The difference is that volumes attached to pods that are part of a stateful set will always be the same (unless they are local volumes). With deployments, a new claim will be made for each new pod.

If the collector is not writing anything to the local storage (as is usually the case, as the collector is typically stateless), the only advantage of stateful sets is the predictable naming.

Going back to your example: are you saying that the Prometheus receiver is locally storing data on disk? I thought it would scrape the endpoints and dispatch the data points through the pipeline.

Reply:
I agree in our case, the StatefulSet Collector doesn't have the advantage in terms of the volumes because it doesn't write data to the local storage. But I think other than predictable names, loadbalancing (design doc [here](https://docs.google.com/document/d/1GdpCqdkz72291l_8S-jio7EhVltminobvqshsFih6vA/edit?usp=sharing), implementation is ongoing) is another advantage which the StatefulSet Collector owns. Maybe I could remove the persistent storage part and give a few descriptions about the loadbalancing advantages here. When this feature is merged upstream, I will add more details.

> are you saying that the Prometheus receiver is locally storing data on disk? I thought it would scrape the endpoints and dispatch the data points through the pipeline.

No, sorry for the confusion. I was trying to give the [Prometheus metrics local storage examples](https://prometheus.io/docs/prometheus/latest/storage/) to better illustrate the idea of the PV. I will rephrase this paragraph to be more clear.